src/plugins/discover/public/url_generator.ts

```diff
@@ -0,0 +1,114 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import {
+  TimeRange,
+  Filter,
+  Query,
+  esFilters,
+  QueryState,
+  RefreshInterval,
+} from '../../data/public';
+import { setStateToKbnUrl } from '../../kibana_utils/public';
+import { UrlGeneratorsDefinition } from '../../share/public';
+
+export const DISCOVER_APP_URL_GENERATOR = 'DISCOVER_APP_URL_GENERATOR';
+
+export interface DiscoverUrlGeneratorState {
+  /**
+   * Optionally set saved search ID.
+   */
+  savedSearchId?: string;
+
+  /**
+   * Optionally set index pattern ID.
+   */
+  indexPatternId?: string;
+
+  /**
+   * Optionally set the time range in the time picker.
+   */
+  timeRange?: TimeRange;
+
+  /**
+   * Optionally set the refresh interval.
+   */
+  refreshInterval?: RefreshInterval;
+
+  /**
+   * Optionally apply filers.
+   */
+  filters?: Filter[];
+
+  /**
+   * Optionally set a query. NOTE: if given and used in conjunction with `dashboardId`, and the
+   * saved dashboard has a query saved with it, this will _replace_ that query.
+   */
+  query?: Query;
```

Comment:
just load thinking 🤔 , could there be a use case where `columns`, `sort` or `interval` (of the histogram) are needed?

Reply:
Possibly, here I just specified options wich already existed in Dashboard URL generator and few extra ones which will enable: https://github.com/elastic/kibana/issues/62652 and https://github.com/elastic/kibana/issues/60227